roles/openshift_grafana/tasks/main.yml

```diff
@@ -2,121 +2,142 @@
 - name: Create grafana namespace
   oc_project:
     state: present
-    name: grafana
+    name: "{{ grafana_namespace }}"
 
 - name: Configure Grafana Permissions
   include_tasks: tasks/gf-permissions.yml
-  when: gf_oauth | default(false) | bool == true
-
-# TODO: we should grab this yaml file from openshift/origin
-- name: Templatize grafana yaml
-  template: src=grafana-ocp.yaml dest=/tmp/grafana-ocp.yaml
-  register:
-    cl_file: /tmp/grafana-ocp.yaml
-  when: gf_oauth | default(false) | bool == false
-
-# TODO: we should grab this yaml file from openshift/origin
-- name: Templatize grafana yaml
-  template: src=grafana-ocp-oauth.yaml dest=/tmp/grafana-ocp-oauth.yaml
-  register:
-    cl_file: /tmp/grafana-ocp-oauth.yaml
-  when: gf_oauth | default(false) | bool == true
+
+- name: Pull grafana yaml from openshift/origin
+  get_url:
+    url: "{{ grafana_base_url }}/grafana.yaml"
+    dest: /tmp/grafana.yaml
+
+- name: Create grafana template
+  oc_obj:
+    namespace: "{{ grafana_namespace }}"
+    kind: template
+    name: grafana
+    state: present
+    files:
+    - "/tmp/grafana.yaml"
 
 - name: Process the grafana file
   oc_process:
-    namespace: grafana
-    template_name: "{{ cl_file }}"
+    namespace: "{{ grafana_namespace }}"
+    template_name: grafana
     create: True
-    when: gf_oauth | default(false) | bool == true
 
 - name: Wait to grafana be running
-  command: oc rollout status deployment/grafana-ocp
+  command: oc rollout status deployment/grafana
 
-- name: oc adm policy add-role-to-user view -z grafana-ocp -n {{ gf_prometheus_namespace }}
+- name: Add view role to grafana user
   oc_adm_policy_user:
-    user: grafana-ocp
+    user: "{{ grafana_user }}"
     resource_kind: cluster-role
     resource_name: view
     state: present
-    role_namespace: "{{ gf_prometheus_namespace }}"
-
-- name: Get grafana route
-  oc_obj:
-    kind: route
-    name: grafana
-    namespace: grafana
-  register: route
-
-- name: Get prometheus route
-  oc_obj:
-    kind: route
-    name: prometheus
-    namespace: "{{ gf_prometheus_namespace }}"
-  register: route
+    role_namespace: "{{ grafana_prometheus_namespace }}"
 
 - name: Get the prometheus SA
-  oc_serviceaccount_secret:
-    state: list
-    service_account: prometheus
-    namespace: "{{ gf_prometheus_namespace }}"
+  shell: oc sa get-token {{ grafana_prometheus_sa }} -n {{ grafana_prometheus_namespace }}
   register: sa
 
-- name: Get the management SA bearer token
-  set_fact:
-    management_token: "{{ sa.results | oo_filter_sa_secrets }}"
-
-- name: Ensure the SA bearer token value is read
-  oc_secret:
-    state: list
-    name: "{{ management_token }}"
-    namespace: "{{ gf_prometheus_namespace }}"
-  no_log: True
-  register: sa_secret
-
-- name: Get the SA bearer token for prometheus
-  set_fact:
-    token: "{{ sa_secret.results.encoded.token }}"
+- name: Get grafana route
+  shell: oc get route grafana -o jsonpath='{.spec.host}' -n {{ grafana_namespace }}
```

Comment:
Use the module for this.
```
oc_route
  state: list
```

Reply:
I tried it already it dosn't work as expected.